ODLA/platforms/odla_popart/test/odla_config_test.cc

```diff
@@ -0,0 +1,216 @@
+      std::string cache_dir="./tmp";
+      CHECK_EQ(odla_SetComputationItem(comp, (odla_item_type) 1001, (odla_item_value)cache_dir.c_str()), ODLA_SUCCESS);
+      
+      //unsuport type 1002 
+      CHECK_EQ(odla_SetComputationItem(comp, (odla_item_type) 1002, nullptr), ODLA_UNSUPPORTED_DATATYPE);
+      CHECK_EQ(odla_DestroyComputation(comp), ODLA_SUCCESS);
+
+    }
+
+    //test odla_SetContextItem ODLA_ASYNC_CALLBACK_ARG, unexpected type
+    {
+      odla_computation comp;
+      CHECK_EQ(ODLA_SUCCESS, odla_CreateComputation(&comp));
+
+      odla_context ctx;
+      odla_CreateContext(&ctx);
+      odla_item_value _test;
+
+      CHECK_EQ(odla_SetContextItem(ctx, ODLA_ASYNC_CALLBACK_ARG, (odla_item_value)&_test), ODLA_SUCCESS);
+      CHECK_EQ(odla_SetContextItem(ctx, (odla_item_type)0, (odla_item_value)&_test), ODLA_UNSUPPORTED_DATATYPE);
+
+      CHECK_EQ(odla_DestroyComputation(comp), ODLA_SUCCESS);
+      CHECK_EQ(odla_DestroyContext(ctx), ODLA_SUCCESS);
+    }
+  }
+
+  SUBCASE("TestMakeNDArrayWrapperType")
+ {
+   odla_void* val;
+   std::vector<int64_t> shape(1, 1); 
+   auto ndarray = MakeNDArrayWrapper(val, popart::DataType::FLOAT16, shape);
+   CHECK_EQ(popart::DataType::FLOAT16, ndarray.get()->dataType());
+
+   ndarray = MakeNDArrayWrapper(val, popart::DataType::UINT32, shape);
+   CHECK_EQ(popart::DataType::UINT32, ndarray.get()->dataType());
+   
+   ndarray = MakeNDArrayWrapper(val, popart::DataType::BOOL, shape);
+   CHECK_EQ(popart::DataType::BOOL, ndarray.get()->dataType());
+
+   ndarray = MakeNDArrayWrapper(val, popart::DataType::INT64, shape);
+   CHECK_EQ(popart::DataType::INT64, ndarray.get()->dataType());
+
+   ndarray = MakeNDArrayWrapper(val, popart::DataType::INT32, shape);
+   CHECK_EQ(popart::DataType::INT32, ndarray.get()->dataType());
+
+   ndarray = MakeNDArrayWrapper(val, popart::DataType::FLOAT, shape);
+   CHECK_EQ(popart::DataType::FLOAT, ndarray.get()->dataType());
+
+ }
+
+  SUBCASE("TestConfigFunctionCallAfterInited")
+   {
+      json config_json = default_json();
+      config_json["execution_mode"] = "parallel";
+      config_json["queue_type"] = "ContextQueues";
+      config_json["queue_capacity"] = 512 * 512;
+
+      std::string path("./TestConfigFunctionCallAfterInited.json");
+      std::ofstream file(path);
+      file << config_json;
+      file.close();
+
+      CHECK_EQ(ODLA_SUCCESS, PopartConfig::instance()->load_config(path.c_str()));
+      CHECK_EQ(true, PopartConfig::instance()->inited());
+      CHECK_EQ(std::string("ContextQueues"), PopartConfig::instance()->queue_type());
+      CHECK_EQ(512 * 512,  PopartConfig::instance()->queue_capacity());
+      CHECK_EQ(ExecutionMode::PARALLEL,  PopartConfig::instance()->execution_mode());
+
+      CHECK_NOTHROW_MESSAGE(PopartConfig::instance()->parse_from_json(config_json),  "config already inited");
+      CHECK_NOTHROW_MESSAGE(PopartConfig::instance()->load_config(path.c_str()),  "config already inited");
+   }
+
+}
+
+
+

```

Comment:
Need to add a function to check the default value of configuration, in case some one change it by mistake.